posts/frangeris/06.stale-cache.md

```diff
@@ -0,0 +1,129 @@
+
+While creating the pull zone, they provide the options to check "in what zones" we want to store the cached data, so the content is served from the closest region instead of traveling all the way to the origin, we will select all the zones available so the cache is everywhere 🤯.
+
+![Pull zone for edge locations](./assets/06.image.4.png)
+
+With the pull zone created, is ready to be used as proxy for the API, now lets move back the the endpoint code to make the header customization so bunny know what to do with our origin responses.
+
+We can start making request to the new domain pull zone domain, so any requests made to CDN will be forwarded behind scenes to API but caching the response.
+
+```plaintext
+// instead of
+GET https://api.example.com/random
+
+// use the pull zone domain
+GET https://mynewpullzone.b-cdn.net/random
+```
+
+By default our cache control header is se to `no-cache`, so every time we hit the pull zone url provided it will make a request to the origin, so lets change that behavior by just adding `max-age` directive to the header.
+
+```plaintext
+"Cache-Control": `public, max-age=86400`,
+```
+
+This will tell the CDN to cache the response for 1 day, so the next requests will be served from the cache instead of making a request to the origin. And by enabling the previous regions zones, the request will be served from the closest edge location to the client getting a lightning fast response.
+
+![Pull zone for edge locations](./assets/06.image.5.png)
+
+🔥 `18ms` of response time, that's a huge improvement from the previous `200ms - 450ms`, and this is just the beginning.
+
+The `CDN-Cache` header returned in the request tell us if the request was served by the CDN cache or not, in case the value is `MISS`, the request was made to the origin, if the value is `HIT`, the request was served from the cache.
+
+There are other features that can be implemented, bunny provide features like "Query String Sorting" that when enabled, the query parameters will be automatically sorted into a consistent order before checking the cache. For example, this will treat `/random?width=200&height=100` and `/random?height=100&width=200` as the same path.
+
+Implementing a cache system is crucial for improving the performance and scalability of your API services. By leveraging the concept of stale cache and using CDNs like Bunny.net, you can significantly reduce response times and server load. The `Cache-Control` header, with directives like `stale-while-revalidate`, allows you to serve stale content while fetching fresh data in the background, ensuring that your users always get the best possible experience. Remember, while Bunny.net is used in this example, these caching principles can be applied with any CDN provider that supports similar capabilities. Happy caching!
```

Comment:
Is this the conclusion? I think you need an extra title.

Reply:
You mean `### Conclusion`?

Reply:
or another title to mark the conclusion.

Reply:
Added